### PR TITLE
Integrate blacklist in no-os build

### PR DESCRIPTION
**Pull request by AAndrisa**
## Pull Request Description

Some projects from hdl are failing, in order to ignore this failing projects on hdl side a 'blacklist' was created.
This list is now processed in no-os to skip the build for those projects as well.
The build branch needs to be exported because of the changes regarding blacklist integration.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [x] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Comment by AAndrisa** [comment]
v1: add blacklist initialization in configure_and_download_all_hw function

File: tools/scripts/build_projects.py [hunk 1]
```diff
@@ -146,6 +150,19 @@ def to_blue(str):
...
 
+def process_blacklist():
+	blacklist = []
+	err = os.system('wget --http-user={} --http-password={} --auth-no-challenge \'{}\' -O blacklist.txt >> {} 2>&1'
+				.format(USER, TOKEN, blacklist_url, log_file))
+	if err != 0 or (not os.path.isfile('blacklist.txt')):
+		log_err('Can not download blacklist file')
+		return blacklist
+	file = open('blacklist.txt', 'r')
+	for line in file.readlines():
```
Comment (SRaus) [inline tools/scripts/build_projects.py:161]: Add  also a condition for empty lines, beside the commented ones.

**Review by SRaus — CHANGES_REQUESTED** [review]

**Comment by AAndrisa** [comment]
v2: added check for empty lines

**Review by SRaus — APPROVED** [review]

**Comment by AAndrisa** [comment]
v3: change comments check in blacklist file

**Review by rbolboac — APPROVED** [review]

**Pull request merged**